packages/vite/src/rsc/rscStudioHandlers.ts

```diff
@@ -0,0 +1,80 @@
+import http from 'node:http'
+import type { PassThrough } from 'node:stream'
+
+export const processRenderRscStream = async (
+  pipeable: PassThrough
+): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const chunks = [] as any
+
+    pipeable.on('data', (chunk: any) => {
+      chunks.push(chunk)
+    })
+
+    pipeable.on('end', () => {
+      const resultBuffer = Buffer.concat(chunks)
+      const resultString = resultBuffer.toString('utf-8') as string
+      resolve(resultString)
+    })
+
+    pipeable.on('error', (error) => {
+      reject(error)
+    })
+  })
+}
+
+export const postFlightToStudio = (
+  payload: string,
+  metadata: Record<string, any>
+) => {
+  const base64Payload = Buffer.from(payload).toString('base64')
+  const encodedMetadata = Buffer.from(JSON.stringify(metadata)).toString(
+    'base64'
+  )
+  const jsonBody = JSON.stringify({
+    flight: {
+      encodedPayload: base64Payload,
```

Comment:
BTW - One reason I am encoded the payload is that it makes it faster to persist in Studio by saving these as strings since Studio SQLite/Prisma lacks native JSON support. I can simply decode on the fetch and not worry about the type conversion and enforce types on the SDL rather than JSON schema or a related model schema. I can validate with zod if need.